### ISSUE TITLE
[featurerequest] add orElse to ifPresent

### ISSUE THREAD

**Issue by MikeMitterer** [tonio-ramirez/optional.dart#2]
Could you please add a "orElse" to ifPresent?

```dart
    void ifPresent(void consume(T val),{ void orElse() }) {
        if(isPresent) {
            consume(_value);
        } else if(orElse != null) {
            orElse();
        }
    }
```
void ifPresent(void consume(T val)) => Function.apply(consume, [_value]);

**Comment by tonio-ramirez** [comment]
Sure... implemented in b5e02a561ed4b88e374cf265a6b25e0c135a4044

**Comment by MikeMitterer** [comment]
Cool + Thanks!
